chart/compass/templates/system-fetcher-configmap.yaml

```diff
@@ -0,0 +1,190 @@
+{{if .Values.global.systemFetcher.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: compass-system-fetcher-templates-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+    release: {{ .Release.Name }}
+data:
+  integration-systems.json: |-
+    [
+      {
+        "id":"{{ (index .Values.global.systemFetcher.intSystems 0).id }}",
+        "name":"sap-s4hana-cloud",
+        "description":"Integration system for SAP S4Hana Cloud applications"
+      },
+      {
+        "id":"{{ (index .Values.global.systemFetcher.intSystems 1).id }}",
+        "name":"sap-successfactors",
+        "description":"Integration system for SAP SuccessFactors"
+      }
+    ]
+  templates.json: |-
+    [
+      {
```

Comment:
There are a lot of duplications and also default properties that we always set to app templates like the following:
- name and description are always the same
- all app templates have the same 2 placeholders - name and display-name
- all of these app templates have the label managed_app_provisioning:false
- all app templates have the legacy field accessLevel: GLOBAL

So it appears that the only dynamic property (which also has very predictable and similar structure and values) is the `applicationInputJSON`.
From this POV would it make sense to you to simplify the `templates.json` to a simpler structure of the like:

```
[
  {
    "name": "SAP ..."
    "applicationInputJSON: "..."
  }
]
```

After that we can programatically add the description, placeholders, labels and accessLevel. Of course it's possible that we may want to be able to specify different values than the defaults that we'll put programatically but we can extend this json array structure as needed

Another benefit of a simpler structure and one that is decoupled from the actual ApplicationTemplate programatic Go structure (as it currently stands this json can be automatically unmarsheled into a Go structure) is that we have freedom in adding additional fields like `integrationSystemName`. This way we can for example solve the issue of not knowing the Integration system ID at design time here, but by knowing the name of the int system we can programatically find it and add it to the applicationInputJSON later at runtime.

Reply:
in my opinion this is not a simplification but a complication. In order to skip a few lines of configuration we will introduce a complex "magic" code.  If we decouple our configuration json from the actual ApplicationTemplate programatic Go structure then we will have: unmarshal to map[string]interface, if exists:=map[key] {<do-smth>}... or we will have to introduce another configuration go struct